borg/archiver.py

```diff
@@ -371,12 +381,11 @@ def do_extract(self, args, repository, manifest, key, archive):
         sparse = args.sparse
         strip_components = args.strip_components
         dirs = []
-        for item in archive.iter_items(lambda item: matcher.match(item[b'path']), preload=True):
+        filter = self.build_filter(matcher, strip_components)
+        for item in archive.iter_items(filter, preload=True):
             orig_path = item[b'path']
             if strip_components:
                 item[b'path'] = os.sep.join(orig_path.split(os.sep)[strip_components:])
-                if not item[b'path']:
-                    continue
```

Comment:
The issue is that if preload is used for getting items then one *has to* retrieve the chunks of every item that was preloaded ( https://github.com/borgbackup/borg/blob/1.0-maint/borg/archive.py#L101 ). Otherwise RemoteRepo keeps sending get()s for chunks that won't be retrieved; the results of those get()s will then fill up the responses-mapping of RemoteRepo.